crates/rpc/rpc/src/eth/core.rs

```diff
@@ -450,9 +453,131 @@ where
     }
 }
 
+/// A helper to build the `EthApi` instance.
+#[derive(Debug)]
+pub struct EthApiBuilder<Provider, Pool, Network, EvmConfig>
+where
+    Provider: BlockReaderIdExt,
+{
```

Comment:
could we move this to builder.rs ? because we likely need a a lot of functions on this one